flang/test/Semantics/OpenMP/ordered01.f90

```diff
@@ -7,6 +7,15 @@ program main
   integer :: i, N = 10
   real :: a, arrayA(10), arrayB(10), arrayC(10)
   real, external :: foo, bar, baz
+ 
+  !ERROR: DEFAULT clause is not allowed on the DO directive
```

Comment:
The error message has to improve.

Reply:
This message is coming from `CheckAllowed` (https://github.com/llvm/llvm-project/blob/cd9a02e2c76ec2f37409c6f7becd61e605c117d8/flang/lib/Semantics/check-directive-structure.h#L487). The message needs to be generic, since this is shared across different directives. Do you have anything specific in mind which we could add? Should we need something specific to default, then it would be better to resort to emitting the error at the position where the previous commit of this PR was doing.

Reply:
Sorry, I was a bit confused by the presence of ordered and was thinking this is a restriction only if ordered is present. I see that this is a general constraint for the DO directive, so it is fine.